common/cpp/react/renderer/components/rnscreens/utils/RectUtil.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <react/renderer/graphics/Float.h>
+#include <react/renderer/graphics/Size.h>
+#include <cmath>
+#include <concepts>
+
+namespace rnscreens {
+
+namespace react = facebook::react;
+
+template <typename T>
```

Comment:
Why did you make it extensible if there is only one usage of the method with exactly the same parameters? Or am I missing something?

Reply:
Just for produced code quality and I'll stand by that. Such util has potential for reusability and it is ready.

Reply:
So when the reusability is needed, you can just change the method, no? What better quality do you introduce with template for only one use case?

Reply:
I'm coming from an angle that it is better to write general and reusable code in principle ([link to C++ core guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#t143-dont-write-unintentionally-non-generic-code) - however this is general principle, not only applying to C++). 
If I'm right: your point is that there is no need here for generalised method, because it is overkill for our current use cases and maybe it introduces higher maintenance costs. I can see your point being valid, but I believe it is the same for my point 😄 

Taking these two into consideration I'll stand by current shape of the implementation, mostly because there is no some metaprogramming-wizadry going on here, it is almost as simple function as it could get. If that was some standard-library-type metaprogramming I think I would give you your due.

Reply:
Note: I don't consider this a blocker, so I'll proceed with merging, we can continue the discussion and take appropriate action if needed later.